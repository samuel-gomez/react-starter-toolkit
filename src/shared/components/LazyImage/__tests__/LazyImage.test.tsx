import { render, waitFor } from '@testing-library/react';
import LasyImage, { loadImage } from '../index';

const defaultProps = { alt: 'alt', name: '' };

describe('<LasyImage />', () => {
  it('Should render LasyImage', async () => {
    const { getByAltText, asFragment } = render(<LasyImage {...defaultProps} name="accordion.svg" alt="alt_accordion" />);
    await waitFor(() => expect(getByAltText('alt_accordion')).toBeDefined());
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render LasyImage unknown image', async () => {
    const { getByAltText, asFragment } = render(<LasyImage {...defaultProps} name="unknown.svg" alt="alt_unknown" />);
    await waitFor(() => expect(getByAltText('alt_unknown')).toBeDefined());
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should render LasyImage when loading', () => {
    const { asFragment } = render(<LasyImage {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('loadImage', () => {
  it('Should return import when loadImage have been called with image name and extension', async () => {
    const result = await loadImage('quality.png');
    expect(result.default).toEqual({
      src: '/img.jpg',
      height: 40,
      width: 40,
      blurDataURL: 'data:image/png;base64,imagedata',
    });
  });

  it('Should return import when loadImage have been called with image name and without extension', async () => {
    const result = await loadImage('text.svg');
    expect(result.default).toEqual({
      src: '/img.jpg',
      height: 40,
      width: 40,
      blurDataURL: 'data:image/png;base64,imagedata',
    });
  });

  it('Should return import when loadImage have been called with image unknown name and without extension', async () => {
    try {
      await loadImage('unknown.svg');
    } catch (error) {
      expect((error as Error).message).toEqual(
        "Cannot find module '../../../../public/images/unknown.svg' from 'src/shared/components/LazyImage/LazyImage.tsx'",
      );
    }
  });
});
