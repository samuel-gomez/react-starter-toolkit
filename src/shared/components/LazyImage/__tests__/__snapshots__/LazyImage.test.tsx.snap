// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LasyImage /> Should render LasyImage 1`] = `
<DocumentFragment>
  <img
    alt="alt_accordion"
    data-nimg="fill"
    decoding="async"
    loading="lazy"
    sizes="(max-width: 768px) 100vw"
    src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
  />
</DocumentFragment>
`;

exports[`<LasyImage /> Should render LasyImage unknown image 1`] = `
<DocumentFragment>
  <img
    alt="alt_unknown"
    data-nimg="fill"
    decoding="async"
    loading="lazy"
    sizes="(max-width: 768px) 100vw"
    src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
  />
</DocumentFragment>
`;

exports[`<LasyImage /> Should render LasyImage when loading 1`] = `
<DocumentFragment>
  <img
    alt="loading..."
    class="af-lasyimage"
    data-nimg="fill"
    decoding="async"
    loading="lazy"
    sizes="100vw"
    src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
  />
</DocumentFragment>
`;
